src/examples/chaining-tx-methods.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  state,
+  State,
+  method,
+  UInt64,
+  PrivateKey,
+  SmartContract,
+  Mina,
+  AccountUpdate,
+} from 'o1js';
+
+class SimpleZkapp extends SmartContract {
+  @state(UInt64) x = State<UInt64>();
+
+  @method
+  async init() {
+    super.init();
+    this.x.set(new UInt64(0));
+  }
+
+  @method
+  async increment() {
+    let x = this.x.get();
+    this.x.requireEquals(x);
+    let newX = x.add(1);
+    this.x.set(newX);
+  }
+}
+
+let Local = Mina.LocalBlockchain({ proofsEnabled: false });
+Mina.setActiveInstance(Local);
+
+// a test account that pays all the fees, and puts additional funds into the zkapp
+let { privateKey: senderKey, publicKey: sender } = Local.testAccounts[0];
+
+// the zkapp account
+let zkappKey = PrivateKey.random();
+let zkappAddress = zkappKey.toPublicKey();
+
+let initialBalance = 10_000_000_000;
+let zkapp = new SimpleZkapp(zkappAddress);
+await SimpleZkapp.analyzeMethods();
+
+const deployTx = Mina.transaction(sender, async () => {
+  let senderUpdate = AccountUpdate.fundNewAccount(sender);
+  senderUpdate.send({ to: zkappAddress, amount: initialBalance });
+  await zkapp.deploy();
+}).sign([senderKey, zkappKey]);
+await deployTx.then((v) => v.prove());
```

Comment:
Was seeing errors throw from within the ocaml bindings whenever I didn't call `prove`, which is strange considering proofs are disabled in this example. Nevertheless, left it in the long-form. Will adjust this depending on whether we tackle #1547.

Reply:
but you can make the example nicer by removing `@method` from `init()`, then this tx won't require a proof